### PR TITLE
ARROW-12332: [Rust] [Ballista] Add simple api server in scheduler

### PR DESCRIPTION
**Pull request by msathis**
Implements GET /executors. We can additional endpoints going forward.

### CODE REVIEW

**Comment by msathis** [comment]
Screenshot 2021-04-12 at 12 22 21 AM

**Comment by andygrove** [comment]
Hi @msathis and thanks for creating the PR. I pulled your branch and tried testing this locally but couldn't get it to work. I'm probably doing something wrong. It would be good if you could add some documentation as part of this PR. Perhaps provide a `curl` command for hitting this API?

**Comment by msathis** [comment]
```
curl --request GET \
  --url http://localhost:50050/executors \
  --header 'Accept: application/json'
```
@andygrove The above curl should work. Added small documentation to the PR as well.

**Comment by Cheappie** [comment]
Hi guys, I don't know how many endpoints you plan to have in future, but did you consider having additional prefix in path like `/scheduler/executors` ? ( FYI, I am just random person that is reading github issues )

**Comment by msathis** [comment]
@Cheappie I don't think we need the prefix since the api is part of the scheduler. But my knowledge is limited.

@andygrove Do you foresee a requirement to support additional resources not related to schedulers in future?

**Review by andygrove — APPROVED** [review]
LGTM. I tested based on the provided curl command:

```
curl --request GET   --url http://localhost:50050/executors   --header 'Accept: application/json'
[{"id":"7b6069c3-b56a-4909-882b-97f103cdecc2","host":"localhost","port":50051}]
```

**Comment by msathis** [comment]
@andygrove Can you please get it merged, if everything is okay?

**Comment by andygrove** [comment]
> Hi guys, I don't know how many endpoints you plan to have in future, but did you consider having additional prefix in path like `/scheduler/executors` ? ( FYI, I am just random person that is reading github issues )

It's a good point but we're so early with this feature it is hard to know how it will evolve. The process is a dedicated scheduler so adding a scheduler prefix seems redundant at this point though.

File: rust/ballista/rust/scheduler/src/main.rs [hunk 1]
```diff
@@ -56,11 +61,35 @@ async fn start_server(
...
+                        );
+                    }
+                    Either::Right(
+                        tonic
+                            .call(req)
+                            .map_ok(|res| res.map(EitherBody::Right))
+                            .map_err(Error::from),
+                    )
+                },
+            ))
```
Comment (edrevo) [inline rust/ballista/rust/scheduler/src/main.rs:91]: Sorry I'm super late to this PR. Does this change mean that a SchedulerGrpcServer is being built for every request?

**Pull request closed without merging**